python/plugins/processing/algs/perl/PerlUtils.py

```diff
@@ -0,0 +1,246 @@
+                    for subfolder in subfolders:
+                        if subfolder.startswith('R-'):
+                            folder = os.path.join(testfolder, subfolder)
+                            break
+                else:
+                    folder = ''
+            else:
+                folder = ''
+
+        return os.path.abspath(unicode(folder))
+
+    @staticmethod
+    def PerlLibs():
+        folder = ProcessingConfig.getSetting(PerlUtils.PERL_LIBS_USER)
+        if folder is None:
+            folder = unicode(os.path.join(userFolder(), 'perllibs'))
+        try:
+            mkdir(folder)
+        except:
+            folder = unicode(os.path.join(userFolder(), 'perllibs'))
+            mkdir(folder)
+        return os.path.abspath(unicode(folder))
+
+    @staticmethod
+    def defaultPerlScriptsFolder():
+        folder = unicode(os.path.join(userFolder(), 'perlscripts'))
+        mkdir(folder)
+        return os.path.abspath(folder)
+
+    @staticmethod
+    def PerlScriptsFolders():
+        folder = ProcessingConfig.getSetting(PerlUtils.PERLSCRIPTS_FOLDER)
+        if folder is not None:
+            return folder.split(';')
+        else:
+            return [PerlUtils.defaultPerlScriptsFolder()]
+
+    @staticmethod
+    def createPerlScriptFromPerlCommands(commands):
+        scriptfile = open(PerlUtils.getPerlScriptFilename(), 'w')
+        for command in commands:
+            scriptfile.write(command + '\n')
+        scriptfile.close()
+
+    @staticmethod
+    def getPerlScriptFilename():
+        return PerlUtils.perlscriptfilename
+
+    @staticmethod
+    def getConsoleOutputFilename():
+        return PerlUtils.getPerlScriptFilename() + '.Perlout'
+
+    @staticmethod
+    def executePerlAlgorithm(alg, progress):
+        # generate new Perl script file name in a temp folder
+        PerlUtils.rscriptfilename = getTempFilenameInTempFolder('processing_script.pl')
+        # run commands
+        PerlUtils.verboseCommands = alg.getVerboseCommands()
+        PerlUtils.createPerlScriptFromPerlCommands(alg.getFullSetOfPerlCommands())
+        if isWindows():
+            if ProcessingConfig.getSetting(PerlUtils.PERL_USE64):
+                execDir = 'x64'
+            else:
+                execDir = 'i386'
+            command = [
+                PerlUtils.PerlFolder() + os.sep + 'bin' + os.sep + execDir + os.sep
+                + 'perl.exe',
+                'CMD',
+                'BATCH',
+                '--vanilla',
+                PerlUtils.getPerlScriptFilename(),
+                PerlUtils.getConsoleOutputFilename()
+            ]
+
+        else:
+            os.chmod(PerlUtils.getPerlScriptFilename(), stat.S_IEXEC | stat.S_IREAD
+                     | stat.S_IWRITE)
+            command = 'PERL CMD BATCH --vanilla ' + PerlUtils.getPerlScriptFilename() \
+                + ' ' + PerlUtils.getConsoleOutputFilename()
+
+        proc = subprocess.Popen(
```

Comment:
For new code, make sense to use QgsTask? what do you think @nyalldawson @volaya @alexbruy ?